### PR TITLE
Update update-user-profile-task.adoc

### PR DESCRIPTION
**Pull request by sarathecoubian**

### CODE REVIEW

File: modules/ROOT/pages/update-user-profile-task.adoc [hunk 1]
```diff
@@ -5,6 +5,6 @@ endif::[]
 
 When you log into the configured IdP for the first time, an Anypoint user is created using your IdP information. 
 
-Your user profile includes a username, your name, and email address. Typically, the OpenID Connect `userinfo` endpoint provides a `preferred_username` claim that contains the authenticated user’s username. However, some IdPs such as OpenAM and PingFederate do not support the `preferred_username` claim. Therefore, when Anypoint Platform creates a user, the `email` and `subclaims` values comprise the username if the `preferred_username` claim isn't present.
+Your user profile includes a username, your name, and email address. Typically, the OpenID Connect `userinfo` endpoint provides a `preferred_username` claim that contains the authenticated user’s username. However, some IdPs such as OpenAM and PingFederate do not support the `preferred_username` claim. Therefore, when Anypoint Platform creates a user, the `email` and `subclaims` values comprise the username if the `preferred_username` claim isn't present. The claims are made in the following order: `preferred_username`, `email`, `sub`.
```
Comment (sarathecoubian) [inline modules/ROOT/pages/update-user-profile-task.adoc]:
```suggestion
Your user profile includes a username, your name, and email address. Typically, the OpenID Connect `userinfo` endpoint provides a `preferred_username` claim that contains the authenticated user’s username. However, some IdPs such as OpenAM and PingFederate do not support the `preferred_username` claim. Therefore, when Anypoint Platform creates a user, the `email` and `subclaims` values comprise the username if the `preferred_username` claim isn't present.  Claims are made in the following order: `preferred_username`, `email`, `sub`. If a username is created as a single string of random characters and does not have an email address defined, the string may be returning the `sub` claim that the id token and userinfo endpoint shows for the user.

File: modules/ROOT/pages/update-user-profile-task.adoc [hunk 2]
```diff
@@ -6,5 +6,6 @@ endif::[]
 When you log into the configured IdP for the first time, an Anypoint user is created using your IdP information. 
 
 Your user profile includes a username, your name, and email address. Typically, the OpenID Connect `userinfo` endpoint provides a `preferred_username` claim that contains the authenticated user’s username. However, some IdPs such as OpenAM and PingFederate do not support the `preferred_username` claim. Therefore, when Anypoint Platform creates a user, the `email` and `subclaims` values comprise the username if the `preferred_username` claim isn't present.
+Claims are made in the following order: `preferred_username`, `email`, `sub`. If a username is created as a single string of random characters and does not have an email address defined, the string may be returning the `sub` claim that the id token and userinfo endpoint shows for the user.
```
Comment (rddheath) [inline modules/ROOT/pages/update-user-profile-task.adoc]: "may be returning": That sounds odd to me. Perhaps "might return"?

File: modules/ROOT/pages/update-user-profile-task.adoc [hunk 3]
```diff
@@ -6,5 +6,6 @@ endif::[]
...
 
 Your user profile includes a username, your name, and email address. Typically, the OpenID Connect `userinfo` endpoint provides a `preferred_username` claim that contains the authenticated user’s username. However, some IdPs such as OpenAM and PingFederate do not support the `preferred_username` claim. Therefore, when Anypoint Platform creates a user, the `email` and `subclaims` values comprise the username if the `preferred_username` claim isn't present.
+Claims are made in the following order: `preferred_username`, `email`, `sub`. If a username is created as a single string of random characters and does not have an email address defined, the string may be returning the `sub` claim that the id token and userinfo endpoint shows for the user. 
 
-To update user profile information from your configured IdP, log into Anypoint Platform through the IdP. Your user profile information is updated within Anypoint Platform using the updates from the IdP.
\ No newline at end of file
+To update user profile information from your configured IdP, log into Anypoint Platform through the IdP. Your user profile information is updated within Anypoint Platform using the updates from the IdP.
```
Comment (rddheath) [inline modules/ROOT/pages/update-user-profile-task.adoc]: "IdP" should be "IDP", right?

Comment (rddheath) [inline modules/ROOT/pages/update-user-profile-task.adoc]: I did not understand this paragraph.
"To update user-profile information from your configured IDP": Do you mean "To update your Anypoint Platform user profile with information from your configured IDP, log into Anypoint Platform through the IDP."?
"Your user-profile information is updated within Anypoint Platform using the updates from the IDP": Perhaps "When you log in, your user profile is automatically updated with the information."?

File: modules/ROOT/pages/update-user-profile-task.adoc (hunk 2 above)
Comment (rddheath) [inline modules/ROOT/pages/update-user-profile-task.adoc]: Isn't every username "a single string of random characters"?

Comment (rddheath) [inline modules/ROOT/pages/update-user-profile-task.adoc]: "that the id token and userinfo endpoint shows":
1. The verb should be "show".
2. How do a token and endpoint "show" something? Do you mean that the claim is part of the value of the token and of the value of the endpoint?

**Review by rddheath — APPROVED** [review]

File: modules/ROOT/pages/update-user-profile-task.adoc (hunk 2 above)
Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]: No. The username is pulled from `preferred_username` or `email` and `subclaims`, as stated above.

Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]: This changes the meaning. The string has already been returned.

Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]:
```suggestion
Claims are made in the following order: `preferred_username`, `email`, `sub`. If a username is created as a single string of random characters and does not have an email address defined, the string may be returning the `sub` claim that the id token and userinfo endpoint show for the user. 
```

File: modules/ROOT/pages/update-user-profile-task.adoc (hunk 3 above)
Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]: No.

Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]: I didn't modify this, so I don't know, but I suppose that works.

Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]:
```suggestion
To update user profile information from your configured IdP, log into Anypoint Platform through the IdP. When you log in, your user profile is automatically updated with the information from the IdP.
```

File: modules/ROOT/pages/update-user-profile-task.adoc (hunk 2 above)
Comment (rddheath) [reply modules/ROOT/pages/update-user-profile-task.adoc]: I didn't get that.

Comment (rddheath) [reply modules/ROOT/pages/update-user-profile-task.adoc]: @sarathecoubian Yeah, but the values of those things must be single strings of random characters.

Comment (sarathecoubian) [reply modules/ROOT/pages/update-user-profile-task.adoc]: They're not random if they're defined.

**Pull request merged**